content/courses/onchain-development/anchor-pdas.md

```diff
@@ -448,40 +453,27 @@ pub mod anchor_movie_review_program {
 }
 
 #[account]
+#[derive(InitSpace)]
 pub struct MovieAccountState {
     pub reviewer: Pubkey,    // 32
     pub rating: u8,          // 1
+    #[max_len(20)]
```

Comment:
Question: does max_len include the 4 bytes? So the string can only be 16 bytes long? Or does it exclude the length storage, so the string can be 20 bytes long? The Anchor docs are vague about it.

Reply:
I suppose its added separately keeping the string 20 chars long. 
As string are actually `Vec<u8>` when serialized, having the length prefix (4 bytes) and the actual string length (20 bytes).

Reply:
and the `#[derive(InitSpace)]` deals with all the space allocation in this case.